controllers/perconaservermysqlbackup_controller.go

```diff
@@ -335,12 +333,23 @@ func (r *PerconaServerMySQLBackupReconciler) getBackupSources(ctx context.Contex
 }
 
 func (r *PerconaServerMySQLBackupReconciler) checkFinalizers(ctx context.Context,
-	cr *apiv1alpha1.PerconaServerMySQLBackup) error {
+	cr *apiv1alpha1.PerconaServerMySQLBackup) {
 	if cr.DeletionTimestamp == nil {
-		return nil
+		return
 	}
 	l := log.FromContext(ctx).WithName("checkFinalizers")
 
+	defer func() {
+		if err := r.Update(ctx, cr); err != nil {
+			l.Error(err, "failed to update finalizers for backup", "backup", cr.Name)
+		}
+	}()
+
+	if cr.Status.State == apiv1alpha1.BackupNew {
```

Comment:
Is it possible to handle this inside case `delete-backup`, just have it somehow in one place? Maybe it's not, didn't look into the details, but if it is it could be good to have everything back up related in one place.

Reply:
In my opinion, we should leave this part for all finalizers. I don't think that any finalizer should run if the backup hasn't ever started